src/Factory/Consignment/ConsignmentFactory.php

```diff
@@ -95,10 +96,15 @@ public function fromOrder(
         $this->setOrderData($order, $deliveryOptions);
         $this->createConsignment();
 
-        $collection  = new ConsignmentCollection();
-        $consignment = $this->initConsignment();
+        $collection        = new ConsignmentCollection();
+        $consignment       = $this->initConsignment();
+        $labelAmount       = $this->request['extraOptions']['labelAmount'] ?? 1;
+        $distributedWeight = $consignment->getTotalWeight() / $labelAmount;
+        $consignment->setTotalWeight($distributedWeight);
 
-        for ($i = 0; $i < ($this->request['extraOptions']['labelAmount'] ?? 1); ++$i) {
+        $this->validateWeight($distributedWeight);
```

Comment:
kan dit niet ook in een validator? eentje die door order->shipments loopt en dan kijkt of alles goed is?